src/features/session_replay/aggregate/index.js

```diff
@@ -121,16 +120,26 @@ export class Aggregate extends AggregateBase {
         }
       }, this.featureName, this.ee)
 
-      this.waitForFlags(['sr']).then(([flagOn]) => this.initializeRecording(
+      Promise.all([
+        this.waitForFlags(['sr']),
+        this.waitForOptIn()
+      ]).then(([[flagOn]]) => this.initializeRecording(
         flagOn,
         Math.random() < getConfigurationValue(this.agentIdentifier, 'session_replay.errorSampleRate'),
         Math.random() < getConfigurationValue(this.agentIdentifier, 'session_replay.sampleRate')
       )).then(() => sharedChannel.onReplayReady(this.mode)) // notify watchers that replay started with the mode
```

Comment:
Is there any chance this could result in an unhandled promise rejection?

Reply:
yes potentially, but this was missed after cloning from the other SR opt in branch and this is no longer needed. will remove